src/Workspaces/Remote/ServiceHub/Services/DiagnosticAnalyzer/DiagnosticComputer.cs

```diff
@@ -36,21 +36,23 @@ internal class DiagnosticComputer
         /// each project in the solution, as the CWT does not seem to drop entries until ForceGC happens, leading to significant memory
         /// pressure when there are large number of open documents across different projects to be analyzed by background analysis.
         /// </summary>
-        private static readonly WeakReference<CompilationWithAnalyzersCacheEntry?> s_compilationWithAnalyzersCache = new(null);
```

Comment:
In my measurements, it seems this weak reference was being dropped many times, leading to cache misses and noticeably slower perf. Given that this cache entry only stores a reference to the last analyzer Project snapshot, I think it is fine to hold a strong reference here.

Reply:
👍

Reply:
documentation needs to be fixed up.

Reply:
i would also be fine with you storing a strong *mapping* from project co compilation-analyzer-with-cache.  it will still all be updated when teh new solution-checksum comes in, right?  but this way, if you're doing a lot of analysis on lots of projects, you get the benefit right?

Reply:
note: this suggestion is optional. i leave it to your best judgement.

Reply:
> i would also be fine with you storing a strong mapping from project co compilation-analyzer-with-cache. 

I will give this a try in a follow-up change after doing measurements.